### PR TITLE
Additions to revert error code string to error object

### PR DESCRIPTION
**Pull request by angelaxu1**

### CODE REVIEW

File: error.go [hunk 1]
```diff
@@ -101,3 +102,46 @@ func IsErrorCode(err error, code int) bool {
...
+
+func StrToError(str string) *Error {
+	strRemaining := strings.TrimLeft(str, "type: ")
+	typ, strRemaining := SplitStrAtWord(strRemaining, "code:")
+	code, message := SplitStrAtWord(strRemaining, "message:")
+	codeInt, _ := strconv.Atoi(code)
+	return &Error{Type: typ, Code: codeInt, Message: message}
+}
+
+func ParseErrorStr(errorStr string) []*Error {
```
Comment (cdworak) [inline error.go]: can we rename from ErrorsFromString(). It follows a similar model of our function naming seen in Hours

File: error.go [hunk 2]
```diff
@@ -101,3 +102,46 @@ func IsErrorCode(err error, code int) bool {
...
+			if before != "" {
+				before += " "
+			}
+			before += w
+		}
+	}
+	return before, after
+}
+
+func StrToError(str string) *Error {
```
Comment (cdworak) [inline error.go]: and then rename this to ErrorFromString()

Comment (cdworak) [reply error.go]: you can also lowercase this function (strToError, or now errorFromString() because it's a helper function and probably shouldn't be needed outside this package)

File: error.go (hunk 1 above)
Comment (cdworak) [reply error.go]: can we change the errorStr parameter name to str (to match the function above it)

File: error.go [hunk 3]
```diff
@@ -101,3 +102,46 @@ func IsErrorCode(err error, code int) bool {
...
+	strRemaining := strings.TrimLeft(str, "type: ")
+	typ, strRemaining := SplitStrAtWord(strRemaining, "code:")
+	code, message := SplitStrAtWord(strRemaining, "message:")
+	codeInt, _ := strconv.Atoi(code)
```
Comment (cdworak) [inline error.go]: hmm we'll probably want to catch that error if the converstion doesn't go through ok (the strconv that is) and return the error

File: error.go [hunk 4]
```diff
@@ -101,3 +102,46 @@ func IsErrorCode(err error, code int) bool {
 	}
 	return false
 }
+
+func SplitStrAtWord(str string, word string) (string, string) {
+	words := strings.Split(str, " ")
```
Comment (cdworak) [inline error.go]: can you put these in a var block? 

var (
words = strings.Split(
found = false
before, after = ""
)

File: error.go [hunk 5]
```diff
@@ -101,3 +102,46 @@ func IsErrorCode(err error, code int) bool {
 	}
 	return false
 }
+
+func SplitStrAtWord(str string, word string) (string, string) {
```
Comment (cdworak) [inline error.go]: you can also lowercase this function because it is a helper (splitStringAtWord)

**Review by cdworak — COMMENTED** [review]
Looks good! Could you add an error_test.go file and add some tests to see if your errors are parsing as expected? 

you can look at the other _test files in yext-go for example but I imagine your tests could look something like this:

test := []struct{
 ErrorMessage string
 Want []*Error
}{
 ErrorMessage: "This is the error message; This is another error message;",
 Want: []*Error{
 [These are the errors you want],

Let me know if you need help!

**Review by cdworak — APPROVED** [review]
This looks great! i can make an item for  you to add tests for next week (if you won't get to it today)

File: error_test.go [hunk 6]
```diff
@@ -54,3 +56,54 @@ func TestIsNotFoundError(t *testing.T) {
...
 	}
 }
+
+func TestErrorsFromString(t *testing.T) {
+	tests := []struct {
+		ErrorMessage string
+		Want         []*Error
+	}{
+		{
+			ErrorMessage: "type: FATAL_ERROR code: 2015 message: Unknown folder; request uuid: 7199948d-9f0d-4649-9625-495b33ad4940",
```
Comment (cdworak) [inline error_test.go]: looks great! can you add one test for the scenario where the error message is empty? I assume it should return nil in that case?
